File: OOP_Lab/Lab_0/lab_0_2.py
class Student:
    name: str
    age: int
    cgpa: float

    def __init__(self, name, age, cgpa):
        self.name = name
        self.age = age
        self.cgpa = cgpa

    def __str__(self):
        return f"{self.name} (Age: {self.age}, CGPA: {self.cgpa})"


if __name__ == "__main__":
    a = Student("Ahmad Shah", 21, 3.2)
    print(a)

    b = Student("Ali", 22, 3.8)
    print(b)

File: OOP_Lab/Lab_0/lab_0_2.cpp
#include<iostream>
#include<cstring>

using namespace std;

struct Student {
    char NAME[50];
    int AGE;
    float CGPA;

    Student() {}

    Student(char n[], int a, float c) {
        strcpy(NAME, n);
        AGE = a;
        CGPA = c;
    }

    void show() {
        cout << "Name: " << NAME << endl;
        cout << "Age: " << AGE << endl;
        cout << "CGPA: " << CGPA << endl;
    }
};

int main() {
    Student s1("Ahmad Shah", 21, 3.2);
    Student s2;
    strcpy(s2.NAME, "Abdullah");
    s2.AGE = 20;
    s2.CGPA = 3.5;
    s1.show();
    s2.show();
    return 0;
}